### PR TITLE
add ecr library usage

### PR DESCRIPTION
**Pull request by adlerhsieh**

### CODE REVIEW

File: src/ecr/ecr.cr [hunk 1]
```diff
@@ -1,3 +1,22 @@
+# ECR is a template engine for embedding Crystal in HTML.
```
Comment (jhass) [inline src/ecr/ecr.cr]: Not quite true, it allows embedding it into arbitrary plaintext, for example it's used in the `crystal init` command.

File: src/ecr/ecr.cr [hunk 2]
```diff
@@ -1,3 +1,22 @@
...
+# 
+# Quick example:
+#
+#     require "ecr"
+#
+#     class Greeting
+#       def initialize(name)
```
Comment (jhass) [inline src/ecr/ecr.cr]: Could short this ti `def initialize(@name)`

**Comment by jhass** [comment]
I think it would be good to point out that the file is read at compile time and embedded into the binary.

**Comment by adlerhsieh** [comment]
Sure, how about I change it into:

```crystal
# ECR is a template engine for embedding Crystal into HTML or arbitrary plain text.
# It is read when compiled, and is embedded into the binary.
# 
# Quick example:
# 
# require "ecr"
#
# class Greeting
#    def initialize(@name)
#    end
#    ecr_file "greeting.ecr"
# end
#
# # greeting.ecr
# <h1>Greeting, <%= @name %>!</h1>
#
# Greeting.new("John")
# #=> <h1>Greeting, John!</h1>
```

**Comment by jhass** [comment]
How about 
> Embedded Crystal (ECR) is a template language for embedding Crystal code into other text, that includes but is not limited to HTML. The template is read and transformed at compile time and then embedded into the binary.

Also some general text about the differences between `<%= %>` and `<% %>` and the available control structures would be nice I think.

**Comment by adlerhsieh** [comment]
I've updated the documentation in the last commit.

**Comment by jhass** [comment]
Thanks!

**Pull request merged**